account_manager/src/wallet/create.rs

```diff
@@ -56,52 +56,65 @@ pub fn cli_app<'a, 'b>() -> App<'a, 'b> {
                     saved at that path. To avoid confusion, if the file does not already \
                     exist it must include a '.pass' suffix.",
                 )
-                .takes_value(true),
+                .action(ArgAction::Set),
         )
         .arg(
-            Arg::with_name(TYPE_FLAG)
+            Arg::new(TYPE_FLAG)
                 .long(TYPE_FLAG)
                 .value_name("WALLET_TYPE")
                 .help(
                     "The type of wallet to create. Only HD (hierarchical-deterministic) \
                             wallets are supported presently..",
                 )
-                .takes_value(true)
-                .possible_values(&[HD_TYPE])
+                .action(ArgAction::Set)
+                .value_parser([HD_TYPE])
                 .default_value(HD_TYPE),
         )
         .arg(
-            Arg::with_name(MNEMONIC_FLAG)
+            Arg::new(MNEMONIC_FLAG)
                 .long(MNEMONIC_FLAG)
                 .value_name("MNEMONIC_PATH")
                 .help(
                     "If present, the mnemonic will be saved to this file. DO NOT SHARE THE MNEMONIC.",
                 )
-                .takes_value(true)
+                .action(ArgAction::Set)
         )
         .arg(
-            Arg::with_name(STDIN_INPUTS_FLAG)
-                .takes_value(false)
-                .hidden(cfg!(windows))
+            Arg::new(STDIN_INPUTS_FLAG)
+                .action(ArgAction::SetTrue)
+                .hide(cfg!(windows))
                 .long(STDIN_INPUTS_FLAG)
                 .help("If present, read all user inputs from stdin instead of tty."),
         )
         .arg(
-            Arg::with_name(MNEMONIC_LENGTH_FLAG)
+            Arg::new(MNEMONIC_LENGTH_FLAG)
                 .long(MNEMONIC_LENGTH_FLAG)
                 .value_name("MNEMONIC_LENGTH")
                 .help("The number of words to use for the mnemonic phrase.")
-                .takes_value(true)
-                .validator(|len| {
-                    match len.parse::<usize>().ok().and_then(|words| MnemonicType::for_word_count(words).ok()) {
-                        Some(_) => Ok(()),
-                        None => Err(format!("Mnemonic length must be one of {}", MNEMONIC_TYPES.iter().map(|t| t.word_count().to_string()).collect::<Vec<_>>().join(", "))),
-                    }
-                })
+                .action(ArgAction::Set)
+                .value_parser(check_mnemonic_length)
                 .default_value("24"),
         )
 }
 
+fn check_mnemonic_length(len: &str) -> Result<String, String> {
```

Comment:
Why declare as separate function? `value_parser` can take a closure, as long as you annotate the type:

```rust
                .value_parser(|len: &str| {
                    match len.parse::<usize>().ok().and_then(|words| MnemonicType::for_word_count(words).ok()) {
                        Some(_) => Ok(len.to_string()),
                        None => Err(format!(
```